pilot/pkg/model/authentication.go

```diff
@@ -312,7 +313,24 @@ func SdsCertificateConfigFromResourceName(resource string) (SdsCertificateConfig
 	} else if strings.HasPrefix(resource, "file-root:") {
 		filesString := strings.TrimPrefix(resource, "file-root:")
 		split := strings.Split(filesString, ResourceSeparator)
-		if len(split) != 1 {
+		if filesString == "system" && len(split) != 1 {
```

Comment:
Does `SdsCertificateConfigFromResourceName` have a unit test? We need some level of testing before it can be merged.